examples/tfengine/gke_cluster.hcl

```diff
@@ -0,0 +1,175 @@
+              {
+                name     = "pods-range"
+                ip_range = "172.16.0.0/14"
+              },
+              {
+                name     = "services-range"
+                ip_range = "172.20.0.0/14"
+              }
+            ]
+          },
+          {
+            name     = "instance-subnet"
+            ip_range = "10.3.0.0/16"
+          }
+        ]
+      }]
+    }
+  }
+}
+
+template "cluster" {
+  recipe_path = "{{$recipes}}/project.hcl"
+  output_path = "./gke_cluster/cluster"
+  data = {
+    project = {
+      project_id = "cluster-project-example"
+      shared_vpc_attachment = {
+        host_project_id = "networks-example-project"
+        subnets = [{
+          name = "gke-subnet"
+        }]
+      }
+    }
+    resources = {
+      gke_clusters = [{
+        name                   = "gke-cluster"
+        network_project_id     = "networks-example-project"
+        network                = "network"
+        subnet                 = "gke-subnet"
+        ip_range_pods_name     = "pods-range"
+        ip_range_services_name = "services-range"
+        master_ipv4_cidr_block = "192.168.0.0/28"
+      }]
+      service_accounts = [{
+        account_id   = "example-sa"
+        description  = "Service Account"
+        display_name = "Service Account"
+      }]
+      compute_instance_templates = [{
+        name_prefix        = "instance-template"
+        network_project_id = "networks-example-project"
+        subnet             = "instance-subnet"
+        service_account    = "$${google_service_account.example_sa.email}"
+        image_family       = "ubuntu-2004-lts"
+        image_project      = "ubuntu-os-cloud"
+
+        instances = [{
+          name = "instance"
+          access_configs = [{
+            nat_ip       = "$${google_compute_address.static.address}"
+            network_tier = "PREMIUM"
+          }]
+        }]
+      }]
+    }
+    terraform_addons = {
+      raw_config = <<EOF
+resource "google_compute_address" "static" {
+  name    = "static-ipv4-address"
+  project = module.project.project_id
+  region  = "us-central1"
+}
+EOF
+    }
+  }
+}
+
+template "kubernetes" {
+  recipe_path = "{{$recipes}}/project.hcl"
+  output_path = "./gke_cluster/kubernetes"
+
+  data = {
+    project = {
+      project_id = "cluster-project-example"
+      exists     = true
+    }
+    resources = {
+      kubernetes_service_accounts = [{
+        name = "ksa-gke"
+        namespace = "example-namespace"
+        google_service_account_email = "example-sa@cluster-project-example.iam.gserviceaccount.com"
+        provider = "gke-alias"
+      }]
+      workload_identity_configurations = [{
+        project_id = "cluster-project-example"
+        google_service_account_id = "example-sa"
+        kubernetes_service_account_name = "ksa-gke"
+        namespace = "example-namespace"
+        cluster_name = "gke-cluster"
+        location = "us-central1"
+      }]
+    }
+    terraform_addons = {
```

Comment:
How come these aren't in the component?

Reply:
The provider is optional and specific to how a solution is implemented.

I think the `namespace` resource can become it's own component. I'll add that as a separate component.

The `google_container_cluster` resource however is not related to this change. Maybe we can somehow integrate that with the `gke_cluster` resource in another PR.